src/CovidAppointmentTable.js

```diff
@@ -65,22 +87,51 @@ export default function CovidAppointmentTable() {
 	const [onlyShowAvailable, setOnlyShowAvailable] = useState(true);
 
 	useEffect(() => {
+		resetData()
+	}, []);
+
+	useEffect(()=> {
+		setLoading(true)
+		setData(data => data.filter(entry => entry.latitude && entry.longitude)
+			.map(entry=> ({...entry, distance: getDistance(currentLocation, {latitude:entry.latitude || 0, longitude: entry.longitude || 0}) * 0.000621371 }))
+			.sort((a,b) => a.distance > b.distance ? 1 : -1))
+		setLoading(false)
+	}, [currentLocation])
+
+	const resetData = () => {
 		fetch("https://mzqsa4noec.execute-api.us-east-1.amazonaws.com/prod").then(
-			async (res) => {
-				const newData = await res.json();
-				setData(JSON.parse(newData.body).results);
-			}
+		async (res) => {
+			const newData = await res.json();
+			setData(JSON.parse(newData.body).results);
+		}
 		);
-	}, []);
+	}
 
 	const formattedData = sortAndFilterData(
 		transformData(data),
 		sortInfo,
 		onlyShowAvailable
 	);
 
+	const getCurrentLocation = () => {
+		setZipcode("")
+		navigator.geolocation.getCurrentPosition((position) => {
+		  setCurrentLocation({latitude: position.coords.latitude, longitude: position.coords.longitude});
```

Comment:
are there other keys besides latitude and longitude? If not, you could just do
```
setCurrentLocation(position.coords);
```

Reply:
There are additional keys such as `accuracy`